### PR TITLE
Replace underscore with hyphen for OKD pod UID

### PR DESCRIPTION
**Pull request by eyalkoren**
Fixes the OKD pod UID discovery error reported through [Discuss](https://discuss.elastic.co/t/infrastructure-ui-kubernetes-apm-traces-integration/207096).

Related to https://github.com/elastic/apm/pull/165

### Checklist

- [x] Implement code
- [x] Add tests
- [x] Update [CHANGELOG.md](CHANGELOG.md)

### CODE REVIEW

**Comment by eyalkoren** [comment]
Yes, that's at least a relevant thing to let others know. 
The thing is that I am not yet sure whether the right approach is what I did (ie - assume that a pod UID should always have hyphens and never underscores) or look for this specific pattern (and leave the option of using an underscore ID with other K8s systems).
Once I have an opinion on that, I will open that issue.

**Comment by felixbarny** [comment]
Actually, this is also relevant for beats. Maybe there's even something in https://github.com/elastic/beats/blob/master/libbeat/common/kubernetes/metadata.go already?

**Comment by eyalkoren** [comment]
@elastic/beats can someone shed some light here- how do you obtain the Kubernetes pod ID? metricbeat seem to be doing it right. Not sure about libbeat. 
We are parsing the `/proc/self/cgroup` file lines for this purpose, but it seems that in some cases the pod UID we get from this source has underscores replacing the hyphens.

**Comment by urso** [comment]
@exekias Can you comment? In some places beats query the pid and then try to parse `/proc/<pid>` instead of reading from `/proc/self`. But not sure about cgroup info here.

**Comment by felixbarny** [comment]
When making changes to the parsing logic, make sure to update https://github.com/elastic/apm/blob/master/docs/agent-development.md#containerkubernetes-metadata

@exekias can you review that spec and make sure it's in sync with what beats does?

**Comment by exekias** [comment]
:wave: As far as I know we don't use cgroup parsing to retrieve Kubernetes Pod UIDs. Both autodiscover and metadata processor rely on Kubernetes API to retrieve the Pod info, so this transformation is not really needed there.

As per the spec, again, we don't do that, but it looks good to me

File: apm-agent-core/src/main/java/co/elastic/apm/agent/impl/payload/SystemInfo.java [hunk 1]
```diff
@@ -191,6 +191,7 @@ SystemInfo parseContainerId(String line) {
                         for (int i = 1; i <= matcher.groupCount(); i++) {
                             String podUid = matcher.group(i);
                             if (podUid != null && !podUid.isEmpty()) {
+                                podUid = podUid.replace('_', '-');
```
Comment (axw) [inline apm-agent-core/src/main/java/co/elastic/apm/agent/impl/payload/SystemInfo.java]: The hyphens are only escaped when using the systemd cgroup driver:

https://github.com/kubernetes/kubernetes/blob/6d0994fa66e34db874e4e26c86b55ba34e75d1c1/pkg/kubelet/cm/cgroup_manager_linux.go#L86

I think we should make this conditional on the path ending with ".scope"

Comment (eyalkoren) [reply apm-agent-core/src/main/java/co/elastic/apm/agent/impl/payload/SystemInfo.java]: Sure thing. See my latest commit.

**Pull request merged**